### PR TITLE
Fix Reseal to actually try to seal

### LINKED ISSUE #157: Inaccurate pipette state after reseal completed
State field in Multipatch changes to Whole Cell instead of "Out" or In Bath to accurately reflect pipette above tissue surface.

### PR DESCRIPTION
**Pull request by outofculture**
Closes #157

### CODE REVIEW

File: acq4/devices/PatchPipette/states.py [hunk 1]
```diff
@@ -1065,9 +1065,10 @@ class PatchPipetteResealState(PatchPipetteState):
         'retractionSpeed': 0.3e-6,
         'resealTimeout': 10 * 60,
         'numTestPulseAverage': 3,
-        'fallbackState': 'whole cell',
+        'fallbackState': 'bath',
```
Comment (campagnola) [inline acq4/devices/PatchPipette/states.py:1068]: 'whole cell' is the correct fallback state -- we are starting from whole cell; if anything unexpected happens while resealing, then we should return to whole cell. Falling back to bath means that a bug in the reseal procedure causes the cell to be instantly killed.

File: acq4/devices/PatchPipette/states.py [hunk 2]
```diff
@@ -1120,7 +1122,7 @@ def run(self):
                 patchrec['resealInitialResistance'] = initialResistance
 
             # check progress on resistance
-            if len(recentTestPulses) > config['numTestPulseAverage']:
+            if len(recentTestPulses) == config['numTestPulseAverage']:
```
Comment (campagnola) [inline acq4/devices/PatchPipette/states.py:1125]: if len(recentTestPulses) == recentTestPulses.maxlen:

File: acq4/devices/PatchPipette/states.py [hunk 3]
```diff
@@ -1265,7 +1267,7 @@ def run(self):
         dev.clean = True
         self.resetPosition()
         dev.newPatchAttempt()
-        return 'out'          
+        return config['fallbackState']
```
Comment (campagnola) [inline acq4/devices/PatchPipette/states.py:1270]: why?

File: acq4/devices/PatchPipette/states.py (hunk 1 above)
Comment (outofculture) [reply acq4/devices/PatchPipette/states.py:1068]: I was told that when reseal fails, the pipette is in the bath, no longer attached to the cell.

File: acq4/devices/PatchPipette/states.py (hunk 3 above)
Comment (outofculture) [reply acq4/devices/PatchPipette/states.py:1270]: fallbackState is 'out', and isn't used ever, so I had just imagined they were meant to go together.

Comment (outofculture) [reply acq4/devices/PatchPipette/states.py:1270]: "isn't used ever"... in this class. the base class sets `self.nextState` to it, so it is still in use there.

File: acq4/devices/PatchPipette/states.py [hunk 4]
```diff
@@ -1041,16 +1041,18 @@ class PatchPipetteResealState(PatchPipetteState):
...
         Rate at which pressure should change during reseal (default is -0.5 kPa / min)
     retractionSpeed : float
-        Speed in m/s to move pipette during retraction (default is 0.3 um / s)
+        Speed in m/s to move pipette during retraction (default is 0.3e-6 (0.3 µm / s))
     resealTimeout : float
         Seconds before reseal attempt exits
     numTestPulseAverage : int
         Number of test pulses to average when measuring resistance
+    slowDetectionThreshold : float
+        Threshold for slow change in pipette resistance (Ohm) to trigger cell detection (default 200 kOhm)
```
Comment (campagnola) [inline acq4/devices/PatchPipette/states.py:1055]: To trigger reseal detection?

Comment (outofculture) [reply acq4/devices/PatchPipette/states.py:1055]: Maybe? I didn't actually grok the code, I just noticed it was used at all but not included in the defaults or docs.

**Pull request closed without merging**